### DISCUSSION TITLE
Anyone used drag knife support in grblHAL?

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/core#614] [Q&A]
I see there is support for a passive drag knife.  Anyone have tips or suggestions about using it?

Also, I don't think a tangential drag knife is supported. (dragknife turned by a stepper/servo motor)  Is that true? I think Mach, UCCNC and LinuxCNC support tangential.  It would need to determine the direction of the cut from the X, Y direction and turn the tangential cutter in that direction. I think that also means the cutter needs a homing sensor and a way to tell grblHAL the orientation of it.

**Comment by jorgerobles** [comment]
Following this thread and enabling 4th axis on DLC32 MAX (https://github.com/grblHAL/core/discussions/711), I'm wondering to use a magnetic encoder ([MT6701](https://uploadcdn.oneyac.com/attachments/files/brand_pdf/magntek/F3/CA/MT6701QT-STD.pdf), [AS5600](https://files.seeedstudio.com/wiki/Grove-12-bit-Magnetic-Rotary-Position-Sensor-AS5600/res/Magnetic%20Rotary%20Position%20Sensor%20AS5600%20Datasheet.pdf#page=22&zoom=100,96,230), etc) with an attiny (maybe even the magnetic encoder itself has an operation mode) to read on position on the homing cycle of A

Caveats: I got no idea if is possible to enable this behavior ONLY in the homing cycle, avoding alarm trigger upon a near 0/360 rotation.

**Comment by phil-barrett** [comment]
I'm not sure it needs any processing power.  Just an index disk with a single trigger point that feeds into the limit/homing input on the motion controller.  Could be HE or optical.  Very minimal electronics.  Ideally, the motion controller would compute the new XY direction and compute the steps needed to point the knife that way (er, point 180 degrees from the direction). 

Yes, the knife axis would need to not have a limit trigger.  Though soft limit could be used. With the knife axis limits being enough to avoid triggering.

**Reply by jorgerobles** [reply]
Checked out MT6701 default behavior as ABZ. It works without further processing, Still have to check if is enough to be read by the A- homing routine.

**Reply by jorgerobles** [reply]
@phil-barrett successfully tested [MT6701](https://www.aliexpress.com/item/1005007470569280.html) using ABZ mode.
Just plug 5v, GND and then Z to IO19 on a tinybee, using `$5=0`
